kernel/src/sched.rs

```diff
@@ -900,4 +732,236 @@ impl Kernel {
 
         (return_reason, time_executed_us)
     }
+
+    #[inline]
+    unsafe fn handle_syscall_fired<P: Platform>(
+        &self,
+        platform: &P,
+        process: &dyn process::ProcessType,
+        syscall: Syscall,
+    ) {
+        process.debug_syscall_called(syscall);
+        // Enforce platform-specific syscall filtering here.
+        //
+        // Before continuing to handle non-yield syscalls
+        // the kernel first checks if the platform wants to
+        // block that syscall for the process, and if it
+        // does, sets a return value which is returned to
+        // the calling process.
+        //
+        // Filtering a syscall (i.e. blocking the syscall
+        // from running) does not cause the process to loose
+        // its timeslice. The error will be returned
+        // immediately (assuming the process has not already
+        // exhausted its timeslice) allowing the process to
+        // decide how to handle the error.
+        match syscall {
+            Syscall::Yield {
+                which: _,
+                address: _,
+            } => {} // Do nothing
+            _ => {
+                // Check all other syscalls for filtering
+                if let Err(response) = platform.filter_syscall(process, &syscall) {
+                    process.set_syscall_return_value(GenericSyscallReturnValue::Failure(response));
+
+                    return;
+                }
+            }
+        }
+
+        // Handle each of the syscalls.
+        match syscall {
+            Syscall::Memop { operand, arg0 } => {
+                let rval = memop::memop(process, operand, arg0);
+                if config::CONFIG.trace_syscalls {
+                    debug!(
+                        "[{:?}] memop({}, {:#x}) = {:?}",
+                        process.appid(),
+                        operand,
+                        arg0,
+                        rval
+                    );
+                }
+                process.set_syscall_return_value(rval);
+            }
+            Syscall::Yield { which, address } => {
+                if config::CONFIG.trace_syscalls {
+                    debug!("[{:?}] yield. which: {}", process.appid(), which);
+                }
+                if which > (YieldCall::Wait as usize) {
+                    // Only 0 and 1 are valid
+                    return;
```

Comment:
```suggestion
                    // Only 0 and 1 are valid. This is an invalid `yield` which we
                    // handle by continuing to execute the process as if the yield
                    // hadn't happened. We cannot (meaningfully) return an error,
                    // because the process cannot differentiate between us (the kernel)
                    // setting the return value, and a potential upcall function
                    // returning a value.
                    return;
```